f3/granite.go

```diff
@@ -424,25 +440,26 @@ func (i *instance) tryPrepare() {
 func (i *instance) beginCommit() {
 	i.phase = COMMIT
 	i.phaseTimeout = i.alarmAfterSynchrony(PREPARE)
-	i.broadcast(COMMIT, i.value, nil)
+	i.broadcast(i.round, COMMIT, i.value, nil)
 }
 
 func (i *instance) tryCommit(round uint32) {
 	// Unlike all other phases, the COMMIT phase stays open to new messages even after an initial quorum is reached,
 	// and the algorithm moves on to the next round.
-	// A subsequent COMMIT message can cause the node to decide, so there is no check on the current phase.
+	// A subsequent COMMIT message can cause the node to terminate, so there is no check on the current phase.
 	committed := i.roundState(round).committed
-	foundQuorum := committed.ListQuorumAgreedValues()
+	foundQuorum := committed.ListStrongQuorumAgreedValues()
 	timeoutExpired := i.host.Time() >= i.phaseTimeout
 
-	if len(foundQuorum) > 0 && !foundQuorum[0].IsZero() {
-		// A participant may be forced to decide a value that's not its preferred chain.
+	if len(foundQuorum) > 0 && !foundQuorum[0].IsZero() && i.phase != DECIDE && i.phase != TERMINATED {
```

Comment:
Both of these new conditions are redundant. They are confusing because the reader is now left working out why they are here, how the code might get into such a state. Please remove.